File: HSGoogleDrivePicker/HSGoogleDrivePicker/GTLDriveFile+HS.m
//
//  GTLDriveFile.m
//  gdrive
//
//  Created by Rob Jonson on 13/10/2015.
//  Copyright © 2015 HobbyistSoftware. All rights reserved.
//

#import "GTLDriveFile+HS.h"
#import <GoogleAPIClient/GTLDrive.h>


@implementation GTLDriveFile (HS_Helpers)

-(BOOL)isFolder
{
    return [self.mimeType isEqualToString:@"application/vnd.google-apps.folder"];
}

-(NSString*)downloadUrl;
{
    if (self.identifier.length == 0)
    {
        return nil;
    }
    
    NSString *urlString = [NSString stringWithFormat:@"https://www.googleapis.com/drive/v3/files/%@?alt=media",
                           self.identifier];
    return urlString;
}

@end

File: HSGoogleDrivePicker/HSGoogleDrivePicker/GTLDriveFile+HS.swift
//  Created by Rob Jonson on 13/10/2015.
//  Copyright © 2015 HobbyistSoftware. All rights reserved.
//

import GoogleAPIClientForREST

extension GTLRDrive_File {
    func isFolder() -> Bool {
        return mimeType == "application/vnd.google-apps.folder"
    }
    
    var downloadUrlString:String? {
        
        guard let identifier = identifier, identifier.count != 0  else {
            return nil
        }

        let urlString = "https://www.googleapis.com/drive/v3/files/\(identifier)?alt=media"
        return urlString
    }
    
    var downloadURL:URL? {
        guard let downloadUrlString = downloadUrlString else {
            return nil
        }
        return URL.init(string: downloadUrlString)
    }
}
